src/request/export/ExportFile.js

```diff
@@ -228,18 +229,26 @@ class ExportFile extends Nimiq.Observable {
     }
 
     /**
-     * Used to set the key if already decrypted elsewhere. This will disable the password requirement
-     * for use cases where the unencrypted key is needed. In case the encrypted key is needed, the password
-     * must also be provided to bypass its requirement.
-     * Set to null to reenable password requirement.
+     * Used to set the key and the password if already decrypted elsewhere.
+     * It will disable the login-file-unlock page entirely and update the progress-indicator.
      * @param {Key?} key
      * @param {string} [password]
      */
     setKey(key, password) {
         this._key = key;
+        this._password = password;
         if (password) {
-            this._passwordBox.hideInput(true);
-            this._password = password;
+            /* eslint-disable no-new */
+            new ProgressIndicator(this.$exportFileIntroPage.querySelector('.progress-indicator'), 2, 1);
+            new ProgressIndicator(this.$downloadFilePage.querySelector('.progress-indicator'), 2, 2);
+            /* eslint-enable no-new */
+            this._keyAlreadySet = true;
```

Comment:
Seems to me the `_keyAlreadySet` variable rather stores if the _password_ was already set, not the key. So how is this variable different from `!!this._password`?

Reply:
It is used to indicate that the passowrd and key were set from outside of ExportFile. this._password is not only set in this method, but in other places inside of ExportFile, too.

Reply:
Ok, but in the way it is currently used to skip the password input, how is it different from `!!this._password`?

Reply:
the password should only be skipped, when the indicators also show it as a 2-step process. That only happens, when the key was decrypted elsewhere (recovery words in remove-key). !!this._password is not sufficient for that as that would lead to the password being skipped when going back to the first page after entering your password in export file, while the indicators still show it as a 3-step process.

Reply:
I'll change it though, making `_keyAlreadySet` obsolete and changing how `_password` is used. Makes more sense I think.

Reply:
Should be much cleaner now